catalog/task/create-pyxis-image/0.1/create-pyxis-image.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-pyxis-image
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that executes a python script to push a container image's metadata to Pyxis
+  params:
+    - name: pyxisUrl
+      type: string
+      description: The URL to the Pyxis API to be used, ending in `/`
+    - name: pyxisSecret
+      type: string
+      description: The kubernetes secret to use to authenticate to Pyxis
+    - name: certified
+      type: string
+      description: If set to true, the image will be marked as certified in its Pyxis entry
+      default: "false"
+    - name: tag
+      type: string
+      description: The tag to use when pushing the container image metadata to Pyxis
+    - name: isLatest
+      type: string
+      description: If set to true, the image will have a latest tag added with its Pyxis entry
+      default: "false"
+    - name: skopeoInspectFile
+      type: string
+      description: The filename of the saved skopeo inspect output stored on the input workspace
+      default: "skopeo-inspect.json"
+  results:
+    - name: containerImageID
+      description: ID of the created entry in Pyxis
+  workspaces:
+    - name: input
+      description: The workspace where the skopeoInspectFile is located
+  steps:
+    - name: create-pyxis-image
+      image:
+        quay.io/hacbs-release/release-utils@sha256:1cb213e052adc0919f0ef1eb7f0fbb0f1a3b87cb98024095d1e1d036e8f5c5ac
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
+      script: |
+        #!/usr/bin/env sh
+
+        if [[ -z "$(params.pyxisUrl)" || -z "$(params.pyxisSecret)" || -z "$(params.certified)" || \
```

Comment:
I thought that this is already checked by Tekton and shows that some parameters are missing after failing. Not the case?

Reply:
I guess it's best to test this in practice - and I will soon be creating my own task and testing it, so I can check then, but it will take a bit still.

But from reading the docs yesterday it seems to me that there is no such thing as a mandatory parameter in Tekton - it will just be empty. I may be wrong though.

Reply:
Hmm, I asked a colleague from my old team and he says that it will fail if you don't define the parameter (and there is no default)... So I guess we better try and see.

Reply:
If it doesn't have a default value, it's mandatory.

```
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  generateName: hello-
spec:
  taskSpec:
    params:
      - name: message
        type: string
    steps:
    - name: default
      image: ubuntu
      script: |
                echo $(params.message)
```
```
kubectl -n damoreno get taskrun        
NAME          SUCCEEDED   REASON                    STARTTIME   COMPLETIONTIME
hello-tfbr6   False       TaskRunValidationFailed   83s         83s
```
```
status:
    completionTime: "2023-01-26T09:14:29Z"
    conditions:
    - lastTransitionTime: "2023-01-26T09:14:29Z"
      message: 'invalid input params for task hello-tfbr6: missing values for these
        params which have no default values: [message]'
```

Reply:
If you define a default and change it to the empty string, will it run with the default instead? I can check, jw if one of you already know

Reply:
If a parameter has a default, it's not mandatory and it will use the default value (empty string).

Reply:
If I set
```
parameters:
- name: foo
  default: bar
```
and pass in `""` as the value of `foo`, does it use `""` or `bar`

Reply:
Thanks for running that test. So then, `""` is a possibility, so we should keep the checks, right?

Reply:
I'd say we don't need those check - an empty string is not the only bad string, you could also supply a bad url or any other bad string. So if we really wanted to validate the inputs it should be something more robust than just checking for empty strings. All in all I'd say we don't need it.

Reply:
Makes sense to me, removed